examples/thirdPartApp/overlays/production/kustomization.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base
+configMapGenerator:
+  - name: example-app-configmap
+    # behavior: merge
+    literals:
+      - DATABASE_URL=production_database_url
+      - API_KEY=production_api_key
+secretGenerator:
+  - name: example-app-secrets
+    # behavior: merge
+    literals:
+      - DB_PASSWORD=production_data_password
+# patches does not working right here, maybe can open an issue.
+# patches:
+#   - patch: |-
+#       apiVersion: apps/v1
+#       kind: Deployment
+#       metadata:
+#         name: example-app
+#       spec:
+#         replicas: 5
+#         template:
+#           spec:
+#             containers:
+#               - name: example-app
+#                 env:
+#                   - name: ENVIRONMENT
+#                     value: "production"
+#                 volumeMounts:
+#                   - name: config-volume
+#                     mountPath: /etc/config
+#                     readOnly: true
+#                   - name: secrets-volume
+#                     mountPath: /etc/secrets
+#                     readOnly: true
+#             volumes:
+#               - name: config-volume
+#                 configMap:
+#                   name: example-app-configmap
+#               - name: secrets-volume
+#                 secret:
+#                   secretName: example-app-secrets
+#     target:
+#       kind: Deployment
+#       name: example-app
+#       version: apps/v1
```

Comment:
Looks like `version` must be set to `v1`. I think you have to write  `apps` at `group` literally if you need.
Please watch here: https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/#patch-using-path-json6902

Reply:
Nice catch